commons-core/src/main/scala/com/avsystem/commons/serialization/json/JsonStringOutput.scala

```diff
@@ -90,7 +100,8 @@ final class JsonStringOutput(builder: JStringBuilder, options: JsonOptions = Jso
       builder.append('"')
       var i = 0
       while (i < binary.length) {
-        builder.append(f"${binary(i) & 0xff}%02x") //JS has signed chars
+        val b = binary(i)
+        builder.append(toHex((b >> 4) & 15)).append(toHex(b & 15))
```

Comment:
Please use hexadecimal notation in bitwise operations, ie. `0x0F` or `0xF` instead of `15`